shell/platform/fuchsia/flutter/tests/fakes/scenic/fake_resources.h

```diff
@@ -0,0 +1,555 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_FUCHSIA_FLUTTER_TESTS_FAKES_SCENIC_FAKE_RESOURCES_H_
+#define FLUTTER_SHELL_PLATFORM_FUCHSIA_FLUTTER_TESTS_FAKES_SCENIC_FAKE_RESOURCES_H_
+
+#include <fuchsia/images/cpp/fidl.h>
+#include <fuchsia/sysmem/cpp/fidl.h>
+#include <fuchsia/ui/composition/cpp/fidl.h>
+#include <fuchsia/ui/gfx/cpp/fidl.h>
+#include <fuchsia/ui/views/cpp/fidl.h>
+#include <lib/fidl/cpp/interface_handle.h>
+#include <lib/fidl/cpp/interface_request.h>
+#include <lib/zx/vmo.h>
+#include <zircon/types.h>
+
+#include <array>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <variant>
+#include <vector>
+
+inline bool operator==(const fuchsia::images::ImageInfo& a,
+                       const fuchsia::images::ImageInfo& b) {
+  return a.height == b.height && a.width == b.width && a.stride == b.stride &&
+         a.tiling == b.tiling && a.transform == b.transform &&
+         a.alpha_format == b.alpha_format && a.pixel_format == b.pixel_format &&
+         a.color_space == b.color_space;
+}
+
+inline bool operator==(const fuchsia::ui::gfx::vec3& a,
+                       const fuchsia::ui::gfx::vec3& b) {
+  return a.x == b.x && a.y == b.y && a.z == b.z;
+}
+
+inline bool operator==(const fuchsia::ui::gfx::BoundingBox& a,
+                       const fuchsia::ui::gfx::BoundingBox& b) {
+  return a.min == b.min && a.max == b.max;
+}
+
+inline bool operator==(const fuchsia::ui::gfx::ViewProperties& a,
+                       const fuchsia::ui::gfx::ViewProperties& b) {
+  return a.downward_input == b.downward_input &&
+         a.focus_change == b.focus_change && a.bounding_box == b.bounding_box &&
+         a.inset_from_min == b.inset_from_min &&
+         a.inset_from_max == b.inset_from_max;
+}
+
+namespace flutter_runner::testing {
+
+// Forward declarations
+template <typename S>
+struct FakeResourceT;
+
+// Unique (within a Session) identifier for a Resource.
+using FakeResourceId = decltype(fuchsia::ui::gfx::CreateResourceCmd::id);
+constexpr FakeResourceId kInvalidFakeResourceId = 0u;
+
+// Tag type for a Resource "state".  The Resource state keeps alive any handles
+// associated with a resource e.g. view token or sysmem token.
+//
+// "snapshot" Resources are generated from "state" Resources during
+// `SceneGraphFromState()` calls.
+//
+// The `FakeSession` stores `FakeResourceT<StateT>` internally.
+struct StateT {
+  using ResourceT = FakeResourceT<StateT>;
+
+  template <typename T>
+  struct HandleT {
+    T value;
+    zx_koid_t koid;
```

Comment:
Perhaps set the primitive type to a known good value, like ZX_KOID_INVALID, to avoid uninitialized-memory bugs.